operator/v1alpha1/types_etcdbackup.go

```diff
@@ -0,0 +1,102 @@
+	// +kubebuilder:validation:Required
+	// +required
+	Spec EtcdBackupSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status EtcdBackupStatus `json:"status"`
+}
+
+type EtcdBackupSpec struct {
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup file would be saved
+	// +kubebuilder:validation:Required
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="pvcName is immutable once set"
+	PVCName string `json:"pvcName"`
+}
+
+const (
+	// Etcd backup is running
+	Running EtcdBackupState = "Running"
+	// Etcd backup is completed
+	Completed EtcdBackupState = "Completed"
+	// Etcd backup failed
+	Failed EtcdBackupState = "Failed"
+	// Etcd backup is pending
+	Pending EtcdBackupState = "Pending"
+)
+
+// etcdBackupState declares valid gathering state types
+// +kubebuilder:validation:Optional
+// +kubebuilder:validation:Enum=Running;Completed;Failed;Pending
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Running' && self == 'Pending')", message="etcdBackupState cannot transition from Running to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Pending')", message="etcdBackupState cannot transition from Completed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Pending')", message="etcdBackupState cannot transition from Failed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Running')", message="etcdBackupState cannot transition from Completed to Running"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Running')", message="etcdBackupState cannot transition from Failed to Running"
+type EtcdBackupState string
+
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.startTime) || has(self.startTime))",message="cannot remove startTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.finishTime) || has(self.finishTime))",message="cannot remove finishTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.etcdBackupState) || has(self.etcdBackupState))",message="cannot remove etcdBackupState attribute from status"
+// +kubebuilder:validation:Optional
+type EtcdBackupStatus struct {
+	// conditions provide details on the status of the gatherer job.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// etcdBackupState reflects the current state of the etcd backup process.
+	// +kubebuilder:validation:Optional
+	// +optional
+	State EtcdBackupState `json:"etcdBackupState,omitempty"`
```

Comment:
Remove.  use a well known condition instead.  Perhaps Complete, Succeeded, and Failed.